src/box/memtx_tree.cc

```diff
@@ -542,27 +570,12 @@ template <bool USE_HINT>							\
 static int									\
 name(struct iterator *iterator, struct tuple **ret)				\
 {										\
-	memtx_tree_t<USE_HINT> *tree =						\
-		&((struct memtx_tree_index<USE_HINT> *)iterator->index)->tree;	\
 	struct tree_iterator<USE_HINT> *it =   					\
 		get_tree_iterator<USE_HINT>(iterator);				\
```

Comment:
LGTM, but when [commit](https://github.com/tarantool/tarantool/pull/7410/commits/c9b393aff0718d2be875c687c1cec9eb27660d0a) will be in master, this variable will be unused, so don't forget to remove it later. (Or you can move that commit to this PR or even merge PRs)